balancer/grpclb/grpclb_test_util_test.go

```diff
@@ -0,0 +1,87 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpclb
+
+import (
+	"net"
+	"sync"
+	"sync/atomic"
+)
+
+type tempError struct{}
+
+func (*tempError) Error() string {
+	return "grpclb test temporary error"
+}
+func (*tempError) Temporary() bool {
+	return true
+}
+
+type restartableListener struct {
+	net.Listener
+
+	closed uint32
```

Comment:
why have a separate atomic when there's a lock, and test is not time sensitive.
Also there's potentially a connection leak, e.g
`Accept` is called ---> `conn, err = l.Listener.Accept()`
Before we append the newly accepted `conn` to `l.conns`, `Close` is called.
And we close all the conns except the new conn just created by `Accept` in `Close`.
When we finally switch back to `Accept`, we append the `conn` to `l.conns`, which won't be closed and thus leaked.

Reply:
It's actually fine if conn is leaked, because listener isn't responsible for the connections. Stop is to stop conns that were accepted, it doesn't care about future connections. I renamed stop() to stopPreviousConns().
Also, Close shouldn't call stop(). 

atomic removed.